### PR TITLE
add zoomToMouse to the ArcRotationCamera

### PR DESCRIPTION
**Pull request by lovettchris**
Zooming to the mouse location when using a mouse wheel is a very standard operation for any 3D tool or game.   This PR adds this as an optional new behavior for ArcRotateCamera as follows:

```typescript

    /**
     * Gets or Set the boolean value that controls whether or not the mouse wheel
     * zooms to the location of the mouse pointer or not.  The default is false.
     */
     @serialize()
     public zoomToMouseLocation = false;
```

I tested it using localDev just so you can see how it works, but I don't expect the localDev changes to be merged as part of this PR.

This is based on this [forum discussion](https://forum.babylonjs.com/t/how-to-set-zooming-based-on-the-position-of-cursor-in-mouse-wheel/13652/16) and [playground prototype](https://playground.babylonjs.com/#0Y0ALJ#118) and we have this open [github issue ](https://github.com/BabylonJS/Babylon.js/issues/9566) to track it.

### CODE REVIEW

File: localDev/index.html [hunk 1]
```diff
@@ -68,15 +68,7 @@
         // Global to simulate PG.
         var engine = null;
 
-        // Allow querystring to navigate easily in debug in local samples.
```
Comment (deltakosh) [inline localDev/index.html:71]: why that change?

File: localDev/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+
+function createScene(engine) {
```
Comment (deltakosh) [inline localDev/index.js:2]: index.js should not be merged into the repo

File: src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts [hunk 3]
```diff
@@ -24,6 +27,30 @@ export class ArcRotateCameraMouseWheelInput implements ICameraInput<ArcRotateCam
...
+     * "1 / camera.radius * wheelPrecisionEaseInRatio;".  Note that wheelPrecision
+     * is used like this: "wheelDelta / (this.wheelPrecision * 40);" which means
+     * the larger the wheelPrecision the smaller the zoom amount.  So by inverting
+     * the camera.radius we are getting smaller wheelPrecision amounts the further you
+     * get away from the object, which results in larger zooming steps.  A
+     * wheelPrecisionEaseInRatio of about 300 seems to work nicely in most cases, but
+     * it does depend on the overall size of your scene.
+     */
+     @serialize()
+     public wheelPrecisionEaseInRatio = 0;
```
Comment (deltakosh) [inline src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts:45]: what is the difference with wheelDeltaPercentage ?

Adding @sebavan  and @PolygonalSun as your solution could be better

File: src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts [hunk 4]
```diff
@@ -117,6 +153,42 @@ export class ArcRotateCameraMouseWheelInput implements ICameraInput<ArcRotateCam
         }
     }
 
+    /**
+     * Update the current camera state depending on the inputs that have been used this frame.
+     * This is a dynamically created lambda to avoid the performance penalty of looping for inputs in the render loop.
+     */
+    public checkInputs(): void {
+        if (this.wheelPrecisionEaseInRatio) {
```
Comment (deltakosh) [inline src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts:161]: maybe just return direct (to remove a block:
if (!this.wheelPrecisionEaseInRatio) {
return;
}

**Comment by sebavan** [comment]
Like @deltakosh  I am wondering the difference in intent with the use of wheelDeltaPercentage ?

File: localDev/index.html (hunk 1 above)
Comment (lovettchris) [reply localDev/index.html]: reverted

File: localDev/index.js (hunk 2 above)
Comment (lovettchris) [reply localDev/index.js]: reverted

File: src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts (hunk 4 above)
Comment (lovettchris) [reply src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts:161]: done.

File: src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts (hunk 3 above)
Comment (lovettchris) [reply src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts]: You're right, wheelDeltaPercentage  works, so I removed wheelPrecisionEaseInRatio

**Review by deltakosh — APPROVED** [review]

**Comment by deltakosh** [comment]
All good thanks!

**Comment by deltakosh** [comment]
Do you mind updating the camera doc (@PirateJC FYI) ?

**Comment by lovettchris** [comment]
@deltakosh, awesome, is this request for me? "Do you mind updating the camera doc" ?  If so can you give some pointers?

**Comment by deltakosh** [comment]
Yes please :) just add your great feature here: https://github.com/BabylonJS/Documentation/blob/master/content/How_To/camera/camera_introduction.md

**Comment by lovettchris** [comment]
Done, see https://github.com/BabylonJS/Documentation/pull/196

**Comment by deltakosh** [comment]
You rock!

**Pull request merged**